manim/typing.py

```diff
@@ -332,7 +332,7 @@
     VMobjects!
 """
 
-Vector2D_Array: TypeAlias = Point2D_Array
+Vector2D_Array: TypeAlias = npt.NDArray[PointDType]
```

Comment:
```suggestion
Vector2D_Array: TypeAlias = npt.NDArray[Vector2D]
```

Reply:
I'm not sure about this change - when I wrote "An array of `Vector2D` objects", I meant something like "if you index a `Vector2D_Array` with a single index, what you get is a `Vector2D`", not _literally_ "this is an `(M,)`-shaped ndarray containing M `Vector2D` objects", which `npt.NDArray[Vector2D]` would seem to imply. Using `npt.NDArray[PointDType]` would refer to any NumPy ndarray whose elements are floats, in this case ndarrays of shape `(M, 2)`.

I'm not sure whether `npt.NDArray[Vector2D]` would be valid or correct in the first place - there can be ndarrays whose dtype is simply `object`, and one could have an `(M,)`-shaped ndarray with M objects - but this is not what is intended here :thinking:

Reply:
Although I would agree with this change in another context, the NumPy documentation seems to suggest using scalar types for dtypes, so `Vector2D` shouldn't be used as the dtype:

https://numpy.org/devdocs/reference/typing.html#numpy.typing.NDArray